packages/reframed/src/reframed.ts

```diff
@@ -456,27 +456,70 @@ function monkeyPatchIFrameEnvironment(
 	// methods to query for elements that can be retargeted into the reframedContainer
 	const domQueryProperties: (keyof Pick<
 		Document,
-		| "querySelector"
-		| "querySelectorAll"
-		| "getElementsByClassName"
-		| "getElementsByTagName"
-		| "getElementsByTagNameNS"
-	>)[] = [
-		"querySelector",
-		"querySelectorAll",
-		"getElementsByClassName",
-		"getElementsByTagName",
-		"getElementsByTagNameNS",
-	];
+		"querySelector" | "querySelectorAll" | "getElementsByClassName"
+	>)[] = ["querySelector", "querySelectorAll", "getElementsByClassName"];
 	for (const queryProperty of domQueryProperties) {
 		Object.defineProperty(iframeDocumentPrototype, queryProperty, {
 			value: function reframedCreateFn() {
+				// This breaks out of the shadowRoot, which dashboard needs for SqLoadingManager
+				if (arguments[0] === ":root") {
+					return mainDocument.querySelector(":root");
+				}
 				// @ts-expect-error WTD?!?
 				return shadowRoot[queryProperty].apply(shadowRoot, arguments);
 			},
 		});
 	}
 
+	iframeWindow.NodeList = mainWindow.NodeList;
```

Comment:
I think we'll need to do this for every DOM-related constructor to allow instanceof checks in the iframe context to succeed.
![Screenshot 2024-08-14 at 5 14 43 PM](https://github.com/user-attachments/assets/677bcb1b-f1d3-40bb-85db-37e0cd3a1a12)